projects/plugins/crm/admin/settings/general.page.php

```
@@ -90,8 +90,9 @@
 		$updatedSettings['showaddress'] = 1;
 	}
 	$updatedSettings['secondaddress'] = 0;
-	if ( isset( $_POST['wpzbscrm_secondaddress'] ) && ! empty( $_POST['wpzbscrm_secondaddress'] ) ) {
-		$updatedSettings['secondaddress'] = 1;
+	if ( isset( $_POST['wpzbscrm_secondaddress'] ) ) {
+		$updatedSettings['secondaddress']       = (int) $_POST['wpzbscrm_secondaddress'] === SETTING_SECOND_ADDRESS_ALWAYS_HIDE ? 0 : 1; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+		$updatedSettings['secondaddresstoggle'] = (int) $_POST['wpzbscrm_secondaddress'] === SETTING_SECOND_ADDRESS_ALLOW_CHANGE_VISIBILITY ? 1 : 0; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
```

I strongly dislike having a single dropdown turn into two individual `tinyint(1)`s/de facto `bool`s. In my opinion, either we should have two different settings (preferable), or 3 different values.

---

Reading these constants over and over, I continue to get confused as to what they refer to.

Even better and simpler would be to simply have two states: always show both addresses vs. hide but show toggle. The link isn't intrusive and we don't need a separate setting just to disable that.

This prevents the need for a migration as well: the new default (on new installs) will be hide but show toggle.